src/ui/public/vislib/lib/axis.js

```diff
@@ -0,0 +1,228 @@
+import d3 from 'd3';
+import _ from 'lodash';
+import $ from 'jquery';
+import ErrorHandlerProvider from 'ui/vislib/lib/_error_handler';
+import AxisTitleProvider from 'ui/vislib/lib/axis_title';
+import AxisLabelsProvider from 'ui/vislib/lib/axis_labels';
+import AxisScaleProvider from 'ui/vislib/lib/axis_scale';
+
+export default function AxisFactory(Private) {
+  const ErrorHandler = Private(ErrorHandlerProvider);
+  const AxisTitle = Private(AxisTitleProvider);
+  const AxisLabels = Private(AxisLabelsProvider);
+  const AxisScale = Private(AxisScaleProvider);
+  const defaults = {
```

Comment:
My thought was that a single `AxisConfig` class could be used across the different `Axis___` classes. I was thinking it would help document all of the options in one place, and also would help pass necessary configuration values into `AxisScale` and such.